scrape_repo_information.py

```diff
@@ -0,0 +1,101 @@
+else:
+    CLIENT_SECRET = ""
+    print("There doesn't seem to be a GitHub Client Secret; the scraping process may not be fast get one here https://github.com/settings/applications/new")
+
+Pair = collections.namedtuple("Pair", "lg1 lg2 last_updated created direction repo stems")
+types = ["trunk", "nursery", "incubator", "staging"]
+
+def print_info(uri):
+    try:
+        dictX = str((urllib.request.urlopen(uri)).read(), "utf-8")
+        tree = xml.fromstring(dictX)
+    except:
+        return -1 # FIXME: error handling
+
+    return len(tree.findall("*[@id='main']/e//l"))
+
+def main():
+    pairs = []
+    params = urllib.parse.urlencode(dict({"client_id": CLIENT_ID, "client_secret": CLIENT_SECRET}))
+    for repo_name in types:
+        types_html_url = "https://api.github.com/repos/apertium/apertium-%s/contents?" % repo_name
+        types_html_data = urllib.request.urlopen(types_html_url + params)
+        types_html = json.loads(types_html_data.read())
+        for type_content in types_html:
+            if "apertium" in type_content["name"] and (type_content["name"].count("-") == 2):
+                direction = ""
+                lang_pair_name = type_content["name"]
+                #getting names
+                _, lg1, lg2 = lang_pair_name.split("-")
+
+                #getting into repository
+                link = "https://api.github.com/repos/apertium/%s/contents?" % lang_pair_name
+                repo_json = json.loads(urllib.request.urlopen(link + params).read())
+                for el in repo_json:
+                    if el["name"] == "modes.xml":
+                        download_url = el["download_url"]
+                        html_for_blob_utf_8 = str((urllib.request.urlopen(download_url)).read(), "utf-8")
+                        tree = xml.fromstring(html_for_blob_utf_8)
+                        for mode in tree:
+                            if lg1+"-"+lg2 == mode.attrib["name"] and "<" not in direction:
+                                direction += "<"
+                            if lg2+"-"+lg1 in mode.attrib["name"] and ">" not in direction:
+                                direction += ">"
+
+                    elif el["name"] == "apertium-%s.%s.dix" % ((lg1+"-"+lg2), (lg1+"-"+lg2)):
+                        stems = print_info(el["download_url"])
+
+                    if direction == "><":
+                        direction = "<>"
+
+                first_page_commits_link = "https://api.github.com/repos/apertium/%s/commits?" % lang_pair_name
+                first_page_commits_resp = urllib.request.urlopen(first_page_commits_link + params)
+                first_page_commits_html = first_page_commits_resp.read()
+                first_page_commit_json = json.loads(first_page_commits_html)
+
+                if first_page_commits_resp.info().get("Link") is not None:
+                    number_of_pages = first_page_commits_resp.info().get("Link").split(',')[1].split('page=')[1].split('>')[0]
```

Comment:
Please add an example of how this looks in a comment here. This parsing is absolutely inscrutable otherwise.